### ISSUE TITLE
Recurring "Warning: Could not fix..." messages from keg_relocate.rb for libraries linked from Anaconda

### ISSUE THREAD

**Issue by flipphillips** [Homebrew/homebrew#44030]
This -seems- like a recent development here, so I may be experiencing some strange side effect of a change in Mac OS developer tools / something similar. 

`keg_relocate.rb` tries to 'fix' the location of dylibs after builds. Today, I was building the PCL point cloud library and received many messages of the form:

```
Changing install name in /usr/local/Cellar/pcl/1.7.2/pcl_viewer.app/Contents/MacOS/pcl_viewer
  from libpcl_io_ply.1.7.dylib
    to /usr/local/Cellar/pcl/1.7.2/lib/libpcl_io_ply.1.7.dylib
Warning: Could not fix libpython2.7.dylib in /usr/local/Cellar/pcl/1.7.2/pcl_viewer.app/Contents/MacOS/pcl_viewer
Warning: Could not fix libxml2.2.dylib in /usr/local/Cellar/pcl/1.7.2/pcl_viewer.app/Contents/MacOS/pcl_viewer
Warning: Could not fix libz.1.dylib in /usr/local/Cellar/pcl/1.7.2/pcl_viewer.app/Contents/MacOS/pcl_viewer
```
Note that I have Anaconda python installed, and thus, `brew doctor` correctly notes the `-config` scripts that are there (see attached Gist). I have tried to build PCL without Anaconda in my `$PATH` but end up with the same results.

It seems that the three above libraries are the main troublemakers, `libpython2.7.dylib, libxml2.2.dylib` which _are_ in the anaconda world and `libz.1.dylib` which isn't.

The big `if` statement that yields the above errors in `keg_relocate` gives this result if it falls through a series of tests, eg a sort of default condition. 

(NB- I am building this on 10.11GM and I realize this is 'unsupported', yet it soon will be and part of what I'm doing requires El Capitan, etc.)

https://gist.github.com/4af6ec8ab8ffafe3ca23

Note that, when I _do_ execute anything built this way, I get runtime version errors, ala:

```
  Reason: Incompatible library version: pcl_grabcut_2d requires version 12.0.0 or later, but libxml2.2.dylib provides version 10.0.0
```

So it appears that it is using the 'older' `xml2.2` from `/usr/lib` instead of the 'right' one in Anaconda-land.

```
eilmer:~> otool -L /usr/lib/libxml2.2.dylib
/usr/lib/libxml2.2.dylib:
	/usr/lib/libxml2.2.dylib (compatibility version 10.0.0, current version 10.9.0)
	/usr/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.5)
	/usr/lib/libicucore.A.dylib (compatibility version 1.0.0, current version 55.1.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1225.1.1)
eilmer:~> otool -L /Users/flip/anaconda/lib/libxml2.
libxml2.2.dylib  libxml2.a        libxml2.dylib    libxml2.la       

eilmer:~> otool -L /Users/flip/anaconda/lib/libxml2.2.dylib 
/Users/flip/anaconda/lib/libxml2.2.dylib:
	libxml2.2.dylib (compatibility version 12.0.0, current version 12.2.0)
	/usr/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.3)
	/usr/lib/libiconv.2.dylib (compatibility version 7.0.0, current version 7.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 111.0.0)
	/usr/lib/libgcc_s.1.dylib (compatibility version 1.0.0, current version 1.0.0)
```
I have, of course, looked through everything I could find with respect to this error, yet have found nothing of any use.

**Comment by tdsmith** [comment]
I'm sorry; this isn't supported. Homebrew supports building against either system python or Homebrew python. Anaconda is very difficult to support because it bundles libraries that conflict with system and Homebrew libraries.

I'd be happy to help troubleshoot the condition where you see problems after anaconda is removed from PATH.

**Comment by flipphillips** [comment]
OK thank you. I understand this in theory. It would be nice if everyone could play together here.

It -was- working previously… which is interesting, but I am happy to try to build again sans anaconda. I wonder if there was a bunch of ‘cruft’ from previous attempts. Again, in theory, you’d think that the relocation code could use the ‘proper’ libraries as specified by the -config universe, but I understand the garden of problems this might bring.


On Sep 16, 2015, at 5:02 PM, tim smith <notifications@github.com<mailto:notifications@github.com>> wrote:


I'm sorry; this isn't supported. Homebrew supports building against either system python or Homebrew python. Anaconda is very difficult to support because it bundles libraries that conflict with system and Homebrew libraries.

I'd be happy to help troubleshoot the condition where you see problems after anaconda is removed from PATH.

—
Reply to this email directly or view it on GitHub<https://github.com/Homebrew/homebrew/issues/44030#issuecomment-140893324>.

-fp
www.skidmore.edu/~flip<http://www.skidmore.edu/~flip>

**Comment by flipphillips** [comment]
I did a remove-everything and re-brewed everything w/o Anaconda and the compiles were successful. Interesting that there is 'cruft' around from compiles w/ different environments. Over the course of a year I went from a beta Xcode to regular Xcode to just using CLT and there were tons of references to the old build-tools-paths still in the various build logs, etc. 

Anyway- it's disappointing that everything can't play together nicely, but understand.
